qt/widgets/common/src/ImageInfoPresenter.cpp

```diff
@@ -43,4 +47,36 @@ void ImageInfoPresenter::setWorkspace(const Mantid::API::Workspace_sptr &workspa
   }
 }
 
+/**
+ * Fill the view table cells using the model
+ * @param info A reference to a collection of header/value pairs
+ */
+void ImageInfoPresenter::fillTableCells(const ImageInfoModel::ImageInfo &info) {
+  if (info.empty())
+    return;
+  int signalIndex = -1;
+
+  const auto itemCount(info.size());
+  m_view->setColumnCount(itemCount);
+  for (int i = 0; i < itemCount; ++i) {
+    const auto name = info.name(i);
+    if (name == "Signal") {
+      signalIndex = i;
+    }
+    auto header = new QTableWidgetItem(name);
+    header->setFlags(header->flags() & ~Qt::ItemIsEditable);
+    m_view->setItem(0, i, header);
+    auto value = new QTableWidgetItem(info.value(i));
+    value->setFlags(header->flags() & ~Qt::ItemIsEditable);
+    m_view->setItem(1, i, value);
+  }
+  if (signalIndex >= 0) {
+    if (!m_showSignal) {
+      m_view->hideColumn(signalIndex);
+    } else {
+      m_view->showColumn(signalIndex);
+    }
+  }
```

Comment:
```suggestion
    if (name == "Signal") {
      if (m_showSignal) {
        m_view->showColumn(i);
      } else {
        m_view->hideColumn(i);
      }
    }
    auto header = new QTableWidgetItem(name);
    header->setFlags(header->flags() & ~Qt::ItemIsEditable);
    m_view->setItem(0, i, header);
    auto value = new QTableWidgetItem(info.value(i));
    value->setFlags(header->flags() & ~Qt::ItemIsEditable);
    m_view->setItem(1, i, value);
  }
```
Given that the Signal check is only going to be run once, it seems like this could be simplified to remove the additional local variable.